nvflare/apis/stream_shareable.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from builtins import dict as StreamMeta
+from typing import Any, Dict, Tuple
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+
+
+class StreamMetaKey:
+    CHANNEL = "__channel__"
```

Comment:
channel/topic are required parameters for all message layers just like destination. They should not be treated as "meta" data. "meta" in my mind is for optional attributes.

Reply:
It should be considered as "streaming context". It contains contextual information about the end-to-end streaming process. Channel and Topic are required.

I'll change to use the term "StreamContext", instead of StreamMeta.